src/components/controls-shape-offset.js

```diff
@@ -0,0 +1,45 @@
+import { CONTROLLER_OFFSETS } from "./hand-controls2.js";
+
+AFRAME.registerComponent("controls-shape-offset", {
+  schema: {
+    additionalOffset: { default: { x: 0, y: -0.03, z: -0.04 } }
+  },
+  init: function() {
+    this.controller = null;
+    this.shapeAdded = false;
+
+    this._handleControllerConnected = this._handleControllerConnected.bind(this);
+    this.el.addEventListener("controllerconnected", this._handleControllerConnected);
+  },
+
+  remove: function() {
+    this.el.removeEventListener("controllerconnected", this._handleControllerConnected);
+  },
+
+  tick: function() {
+    if (!this.shapeAdded && this.controller) {
```

Comment:
this runs in tick because it needs to run after the `body` component's init has finished (but we also need to have the controllerconnected listener on init or otherwise you'll miss the event if your controllers are already connected when you start the app).